utils/scapp.h

```diff
@@ -38,142 +40,304 @@ struct clampedOption
+                          {"4x3x3", KTX_PACK_ASTC_BLOCK_DIMENSION_4x3x3},
+                          {"4x4x3", KTX_PACK_ASTC_BLOCK_DIMENSION_4x4x3},
+                          {"4x4x4", KTX_PACK_ASTC_BLOCK_DIMENSION_4x4x4},
+                          {"5x4x4", KTX_PACK_ASTC_BLOCK_DIMENSION_5x4x4},
+                          {"5x5x4", KTX_PACK_ASTC_BLOCK_DIMENSION_5x5x4},
+                          {"5x5x5", KTX_PACK_ASTC_BLOCK_DIMENSION_5x5x5},
+                          {"6x5x5", KTX_PACK_ASTC_BLOCK_DIMENSION_6x5x5},
+                          {"6x6x5", KTX_PACK_ASTC_BLOCK_DIMENSION_6x6x5},
+                          {"6x6x6", KTX_PACK_ASTC_BLOCK_DIMENSION_6x6x6}};
+
+  auto opt = astc_blocks_mapping.find(block_size);
+
+  if (opt != astc_blocks_mapping.end())
+      return opt->second;
+
+  return KTX_PACK_ASTC_BLOCK_DIMENSION_6x6;
+}
+
+/**
+ * @memberof ktxTexture
+ * @ingroup write
+ * @~English
+ * @brief Creates valid ASTC quality from string.
+ *
+ * @return      Valid ktx_pack_astc_quality_e from string
+ */
+ktx_pack_astc_quality_levels_e
+astcQualityLevel(const char *quality) {
+
+    static std::unordered_map<std::string,
+                              ktx_pack_astc_quality_levels_e> astc_quality_mapping{
+        {"fastest", KTX_PACK_ASTC_QUALITY_LEVEL_FASTEST},
+        {"fast", KTX_PACK_ASTC_QUALITY_LEVEL_FAST},
+        {"medium", KTX_PACK_ASTC_QUALITY_LEVEL_MEDIUM},
+        {"thorough", KTX_PACK_ASTC_QUALITY_LEVEL_THOROUGH},
+        {"exhaustive", KTX_PACK_ASTC_QUALITY_LEVEL_EXHAUSTIVE}
+    };
+
+  auto opt = astc_quality_mapping.find(quality);
+
+  if (opt != astc_quality_mapping.end())
+      return opt->second;
+
+  return KTX_PACK_ASTC_QUALITY_LEVEL_MEDIUM;
+}
+
+/**
+ * @memberof ktxTexture
+ * @ingroup write
+ * @~English
+ * @brief Creates valid ASTC mode from string.
+ *
+ * @return      Valid ktx_pack_astc_mode_e from string
+ */
+ktx_pack_astc_encoder_mode_e
+astcEncoderMode(const char* mode) {
+    if (strcmp(mode, "ldr") == 0)
+        return KTX_PACK_ASTC_ENCODER_MODE_LDR;
+    else if (strcmp(mode, "hdr") == 0)
+        return KTX_PACK_ASTC_ENCODER_MODE_HDR;
+
+  return KTX_PACK_ASTC_ENCODER_MODE_DEFAULT;
+}
+
 /*
 // Markdown doesn't work in files included by snipped{doc} or include{doc}
 // so the table below has to be laboriously done in html.
 //! [scApp options]
   <dl>
-    <dt>--bcmp</dt>
-    <dd>Supercompress the image data with ETC1S / BasisLZ. Implies @b --t2.
-        RED images will become RGB with RED in each component. RG images will
-        have R in the RGB part and G in the alpha part of the compressed
-        texture. When set, the following BasisLZ-related options become valid
-        otherwise they are ignored.
-      <dl>
+  <dt>--encode &lt;astc|etc1s|uastc&gt;</dt>
+               <dd>Compress the image data to ASTC, transcodable ETC1S / BasisLZ or
+               high-quality transcodable UASTC format. Implies @b --t2.
```

Comment:
I have gone back and forth on this. There is no hard requirement for `--t2`. One should be able to create `.ktx` files with `astc` as well. 

Unfortunatley I couldn't make this to work when I just use a `ktxtexture` instead of `ktxtexture2`. Thats why I have this [FIXME](https://github.com/wasimabbas-arm/KTX-Software/blob/astc_support/lib/astc_encode.cpp#L541). I get the following errors when I use `ktxtexture`.

```
 /System/Volumes/Data/development/KTX-Software/lib/astc_encode.cpp:553:15: no member named 'supercompressionScheme' in 'ktxTexture'

    if (This->supercompressionScheme != KTX_SS_NONE)
        ~~~~  ^
** BUILD FAILED **
The following build commands failed:
❌  /System/Volumes/Data/development/KTX-Software/lib/astc_encode.cpp:564:33: no member named 'pDfd' in 'ktxTexture'
    const uint32_t* BDB = This->pDfd+1;
                          ~~~~  ^
	CompileC /System/Volumes/Data/development/KTX-Software/build-macos-universal/KTX-Software.build/Debug/ktx.build/Objects-normal/x86_64/astc_encode.o /System/Volumes/Data/development/KTX-Software/lib/astc_encode.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)

❌  /System/Volumes/Data/development/KTX-Software/lib/astc_encode.cpp:567:39: no member named 'pDfd' in 'ktxTexture'
    getDFDComponentInfoUnpacked(This->pDfd, &num_components, &component_size);
                                ~~~~  ^
❌  /System/Volumes/Data/development/KTX-Software/lib/astc_encode.cpp:676:29: no matching function for call to 'ktxTexture2_levelDataOffset'
        ktx_size_t offset = ktxTexture2_levelDataOffset(This, level);
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
❌  /System/Volumes/Data/development/KTX-Software/lib/astc_encode.cpp:749:11: no member named 'vkFormat' in 'ktxTexture'
    This->vkFormat = vkFormat;
             ^
❌  /System/Volumes/Data/development/KTX-Software/lib/astc_encode.cpp:751:11: no member named 'supercompressionScheme' in 'ktxTexture'
    This->supercompressionScheme = KTX_SS_NONE;
    ~~~~  ^
❌  /System/Volumes/Data/development/KTX-Software/lib/astc_encode.cpp:752:11: no member named '_private' in 'ktxTexture'
    This->_private->_requiredLevelAlignment = protoPriv._requiredLevelAlignment;
    ~~~~  ^
```

Reply:
To resolve these you would have to

1. Change the function from ktxTexture2 to ktxTexture
2. Check the type of the incoming texture, using its `classId` field, and if t2 make a t2 pointer to it and use that for the checks and operations that are failing.

However I don't want to encourage use of KTX v1, as there is no advantage or compatibility reason that I can see, so let's leave this as a KTX v2 only.